src/main/java/org/apache/sysml/hops/OptimizerUtils.java

```diff
@@ -110,8 +110,13 @@
 	 */
 	public static boolean ALLOW_CONSTANT_FOLDING = true;
 	
-	public static boolean ALLOW_ALGEBRAIC_SIMPLIFICATION = true; 
-	public static boolean ALLOW_OPERATOR_FUSION = true; 
+	public static boolean ALLOW_ALGEBRAIC_SIMPLIFICATION = true;
+	/**
+	 * Enables rewriting chains of element-wise multiplies that contain the same multiplicand more than once, as in
+	 * `A*B*A ==> (A^2)*B`.
+	 */
+	public static boolean ALLOW_EMULT_CHAIN_REWRITE = true;
```

Comment:
Please don't introduce a new constant here but use the existing ALLOW_SUM_PRODUCT_REWRITES (which applies to a number of other rewrites and operators such as mmchain and tak+ as well)